document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java

```diff
@@ -55,81 +50,211 @@
  *                 AggregateExpression.of(SUM, IdentifierExpression.of("col5")),
  *                 GTE,
  *                 ConstantExpression.of(100)))
- *         .sortingDefinition(
+ *         .sort(
  *             FunctionExpression.builder()
  *                 .operand(IdentifierExpression.of("col7"))
  *                 .operator(ADD)
  *                 .operand(IdentifierExpression.of("col8"))
  *                 .build(),
  *             DESC)
- *         .paginationDefinition(PaginationDefinition.of(10, 5))
+ *         .offset(5)
+ *         .limit(10)
  *         .build();
  *  </code>
  */
-@Value
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
-@Builder
-public class Query {
+public final class Query {
```

Comment:
We can change the following as discussed :
- Method names to something like addSelection, setFilter etc to explicitly convey that one adds to a list and the other can reset the previously set value. 
- We can remove filters() method to avoid all error checking to support this
- Intermediate Selection type class added may not be required 
- For consistency we can have addSelection(selection) and addSelection(List<selection>)

Reply:
Instead of 
`Intermediate Selection type class added may not be required`, exposed the definition classes outside as discussed.

The remaining 3 items are done.